src/risedevtool/src/task/meta_node_service.rs

```diff
@@ -77,36 +77,52 @@ impl MetaNodeService {
-        {
-            is_persistent_meta_store = true;
-            let prefix_data = env::var("PREFIX_DATA")?;
-            let file_path = PathBuf::from(&prefix_data)
-                .join(&sqlite_config[0].id)
-                .join(&sqlite_config[0].file);
-            cmd.arg("--backend")
-                .arg("sql")
-                .arg("--sql-endpoint")
-                .arg(format!("sqlite://{}?mode=rwc", file_path.display()));
-        } else {
-            match config.provide_etcd_backend.as_ref().unwrap().as_slice() {
-                [] => {
-                    cmd.arg("--backend").arg("mem");
-                }
-                etcds => {
-                    is_persistent_meta_store = true;
-                    cmd.arg("--backend")
-                        .arg("etcd")
-                        .arg("--etcd-endpoints")
-                        .arg(
-                            etcds
-                                .iter()
-                                .map(|etcd| format!("{}:{}", etcd.address, etcd.port))
-                                .join(","),
-                        );
-                }
+        match (
+            config.provide_etcd_backend.as_ref().unwrap().as_slice(),
+            config.provide_sqlite_backend.as_ref().unwrap().as_slice(),
+            config.provide_postgres_backend.as_ref().unwrap().as_slice(),
+        ) {
+            (etcd_config, [], []) if !etcd_config.is_empty() => {
+                is_persistent_meta_store = true;
+                cmd.arg("--backend")
+                    .arg("etcd")
+                    .arg("--etcd-endpoints")
+                    .arg(
+                        etcd_config
+                            .iter()
+                            .map(|etcd| format!("{}:{}", etcd.address, etcd.port))
+                            .join(","),
+                    );
+            }
+            ([], sqlite_config, []) if !sqlite_config.is_empty() => {
+                is_persistent_meta_store = true;
+                let prefix_data = env::var("PREFIX_DATA")?;
+                let file_path = PathBuf::from(&prefix_data)
+                    .join(&sqlite_config[0].id)
+                    .join(&sqlite_config[0].file);
+                cmd.arg("--backend")
+                    .arg("sql")
+                    .arg("--sql-endpoint")
+                    .arg(format!("sqlite://{}?mode=rwc", file_path.display()));
+            }
+            ([], [], pg_config) if !pg_config.is_empty() => {
+                is_persistent_meta_store = true;
+                cmd.arg("--backend")
+                    .arg("sql")
+                    .arg("--sql-endpoint")
+                    .arg(format!(
+                        "postgres://{}:{}@{}:{}/{}",
+                        pg_config[0].user,
+                        pg_config[0].password,
+                        pg_config[0].address,
+                        pg_config[0].port,
+                        pg_config[0].database
```

Comment:
Since PG can be both CDC and meta backend. I'm thinking whether it will be problematic if we share the user and database for both cases. Ideally sth like `meta_databse` may be better, but it's also ok to me to go with the simple solution and do it later.

Reply:
Sharing the same user and database won't be problematic but requires some extra config for CDC like `wal_level`. Using separate database is definitely much better, but seems like we can have only one initial database via env `POSTGRES_DB`.

Reply:
FYI in this test, the mysql database and user for CDC is created adhoc, instead of specified in `risedev.yml`. 😄 

https://github.com/risingwavelabs/risingwave/blob/07fc213b160bf4d092754eda5bc8e90fb69113c1/e2e_test/source_inline/cdc/mysql/mysql_create_drop.slt#L8-L35

Reply:
Only providing a super user might be good enough

Reply:
Cool! Then there's no problem.